test/e2e/tests/settings-search.spec.js

```diff
@@ -0,0 +1,347 @@
+const { strict: assert } = require('assert');
+const { convertToHexValue, withFixtures } = require('../helpers');
+
+describe('Settings Search', function () {
+  const ganacheOptions = {
+    accounts: [
+      {
+        secretKey:
+          '0x7C9529A67102755B7E6102D6D950AC5D5863C98713805CEC576B945B15B71EAC',
+        balance: convertToHexValue(25000000000000000000),
+      },
+    ],
+  };
+  const settings = {
+    general: {
+      urlSubstring: 'general',
+      target: 'Primary Currency',
+    },
+    advanced: {
+      urlSubstring: 'advanced',
+      target: 'State Logs',
+    },
+    contacts: {
+      urlSubstring: 'contact-list',
+      target: 'Contacts',
+    },
+    security: {
+      urlSubstring: 'security',
+      target: 'Reveal Secret',
+    },
+    alerts: {
+      urlSubstring: 'alerts',
+      target: 'Browsing a website',
+    },
+    networks: {
+      urlSubstring: 'networks',
+      target: 'Ethereum Mainnet',
+    },
+    experimental: {
+      urlSubstring: 'experimental',
+      target: 'Token Detection',
+    },
+    about: {
+      urlSubstring: 'about-us',
+      target: 'Terms of Use',
+    },
+  };
+  let found;
+  let url;
+
+  it('should find element inside the General tab', async function () {
+    await withFixtures(
+      {
+        dapp: true,
+        fixtures: 'imported-account',
+        ganacheOptions,
+        title: this.test.title,
+      },
+      async ({ driver }) => {
+        await driver.navigate();
+        await driver.fill('#password', 'correct horse battery staple');
+        await driver.press('#password', driver.Key.ENTER);
+
+        await driver.clickElement('.account-menu__icon');
+        await driver.clickElement({ text: 'Settings', tag: 'div' });
+        await driver.fill('#search-settings', settings.general.target);
+
+        // Check if element was found
+        found = await driver.isElementPresent({ text: 'General', tag: 'span' });
+        assert.equal(found, true, 'Element was not found');
+
+        // Check if element redirects to the correct page
+        await driver.clickElement({ text: 'General', tag: 'span' });
```

Comment:
Do you need to test for the existence of the element above? `clickElement` will already throw if the element can't be found.

Reply:
That's true. I opted to make an explicit assertion, since this was the only assertion for each "it" block. A commit later I added some extra step and another assertion, so I will remove these ones, as they are already implicit on `clickElement`